core/templates/components/summary-tile/lesson-card.component.ts

```diff
@@ -0,0 +1,153 @@
+import {CollectionSummary} from 'domain/collection/collection-summary.model';
+import {LearnerExplorationSummary} from 'domain/summary/learner-exploration-summary.model';
+import {StorySummary} from 'domain/story/story-summary.model';
+import {StoryNode} from 'domain/story/story-node.model';
+
+interface LessonUrl {
+  classroom: string | undefined;
+  topicFragment: string | undefined;
+  story: string;
+  currentStory: StoryNode;
+}
+
+interface ThumbnailUrl {
+  filename: string;
+  id: string;
+}
+@Component({
+  selector: 'lesson-card',
+  templateUrl: './lesson-card.component.html',
+})
+export class LessonCardComponent implements OnInit {
+  @Input() story!: StorySummary | LearnerExplorationSummary | CollectionSummary;
+  @Input() topic!: string;
+
+  desc!: string;
+  imgColor!: string;
+  imgUrl!: string;
+  lessonUrl!: string;
+  progress!: number;
+  title!: string;
+  lessonTopic!: string;
+
+  constructor(
+    private urlInterpolationService: UrlInterpolationService,
+    private assetsBackendApiService: AssetsBackendApiService
+  ) {}
+
+  ngOnInit(): void {
+    if (this.story instanceof StorySummary) {
+      let completedStories = this.story.getCompletedNodeTitles().length;
+
+      this.desc = this.story.getTitle();
+      this.imgColor = this.story.getThumbnailBgColor();
+      this.imgUrl = this.getStorySummaryThumbnailUrl({
+        filename: this.story.getThumbnailFilename(),
+        id: this.story.getId(),
+      });
+
+      let nextStory =
+        completedStories === this.story.getAllNodes().length
+          ? completedStories - 1
+          : completedStories;
+      /* Last completed story index works because if 1 is completed, 1 index is 2nd item */
+      let lessonArgs = {
+        classroom: this.story.getClassroomUrlFragment(),
+        topicFragment: this.story.getTopicUrlFragment(),
+        currentStory: this.story.getAllNodes()[nextStory],
+        story: this.story.getUrlFragment(),
+      };
+      this.lessonUrl = this.getStorySummaryLessonUrl(lessonArgs);
+
+      this.title = `Chapter ${nextStory + 1}: ${this.story.getNodeTitles()[nextStory]}`;
+      this.progress = Math.floor(
+        (completedStories / this.story.getNodeTitles().length) * 100
+      );
+      this.lessonTopic = this.topic;
+    } else {
+      /* Implementation for explorations & collections */
+      this.desc = this.story.objective;
+      this.imgColor = this.story.thumbnailBgColor;
+      this.imgUrl = this.urlInterpolationService.getStaticImageUrl(
+        this.story.thumbnailIconUrl
+      );
+      this.progress = 0;
+      this.title = this.story.title;
+
+      if (this.story instanceof CollectionSummary) {
+        this.lessonUrl = '';
+        this.lessonTopic = 'Collections';
+      } else {
+        this.lessonUrl = `/explore/${this.story.id}`;
+        this.lessonTopic = 'Community Lessons';
+      }
+    }
+  }
+
+  getProgress(arg: number): string {
```

Comment:
Can you please demonstrate what this function does for different percentages? Maybe there's a simpler way to do what you are doing here?

Reply:
This is for the progress circle. There's a lot of ways on how to do this but they're all rather complicated. If there's a preferred way, I can change it.

Reply:
It's such a basic thing that I would expect to be an existing solution for this
1. I see there's at least a progress circle in story-summary-tile.components.ts, maybe you could extract that code and reuse it
2. If it's difficult, find a decent library to do it

Reply:
Done. Added library.